File: Kotlin/0001.kt
fun sumOfMultiples(limit: Int): Int {
    return (1 until limit)
        .filter { it % 3 == 0 || it % 5 == 0 }
        .sum()
}

fun main() {
    val limit = 1000
    val result = sumOfMultiples(limit)
    println("The sum of all multiples of 3 or 5 below $limit is: $result")
}

File: Scala/0001.scala
object SumMultiples {
  def sumOfMultiples(limit: Int): Int = {
    (1 until limit).filter(n => n % 3 == 0 || n % 5 == 0).sum
  }

  def main(args: Array[String]): Unit = {
    val limit = 1000
    val result = sumOfMultiples(limit)
    println(s"The sum of all multiples of 3 or 5 below $limit is: $result")
  }
}
